packages/lib-sourcify/test/Verification/Verification.spec.ts

```diff
@@ -0,0 +1,1213 @@
+import { describe, it, before, after } from 'mocha';
+import { expect } from 'chai';
+import { Verification } from '../../src/Verification/Verification';
+import SourcifyChain from '../../src/SourcifyChain';
+import { ChildProcess } from 'child_process';
+import { JsonRpcSigner } from 'ethers';
+import path from 'path';
+import {
+  deployFromAbiAndBytecode,
+  expectVerification,
+  vyperCompiler,
+} from '../utils';
+import {
+  startHardhatNetwork,
+  stopHardhatNetwork,
+} from '../hardhat-network-helper';
+import { SolidityMetadataContract } from '../../src/Validation/SolidityMetadataContract';
+import {
+  ISolidityCompiler,
+  SolidityOutput,
+} from '../../src/Compilation/SolidityTypes';
+import { useSolidityCompiler } from '../compiler/solidityCompiler';
+import { findSolcPlatform } from '../compiler/solidityCompiler';
+import fs from 'fs';
+import { VyperCompilation } from '../../src/Compilation/VyperCompilation';
+import { PathContent } from '../../src/Validation/ValidationTypes';
+
+class TestSolidityCompiler implements ISolidityCompiler {
+  async compile(
+    version: string,
+    solcJsonInput: any,
+    forceEmscripten = false,
+  ): Promise<SolidityOutput> {
+    return useSolidityCompiler(version, solcJsonInput, forceEmscripten);
+  }
+}
+
+// Helper function to get compilation from metadata
+async function getCompilationFromMetadata(contractFolderPath: string) {
+  // Read metadata.json directly
+  const metadataPath = path.join(contractFolderPath, 'metadata.json');
+  const metadataRaw = fs.readFileSync(metadataPath, 'utf8');
+  const metadata = JSON.parse(metadataRaw);
+
+  // Read source files from the sources directory
+  const sourcesPath = path.join(contractFolderPath, 'sources');
+  const sources: PathContent[] = [];
+
+  // Get all source files from metadata
+  for (const [sourcePath] of Object.entries(metadata.sources)) {
+    // Extract filename from source path (e.g. "contracts/Storage.sol" -> "Storage.sol")
+    const fileName = sourcePath.split('/').pop() || sourcePath;
+    const filePath = path.join(sourcesPath, fileName);
+    const content = fs.readFileSync(filePath, 'utf8');
+    sources.push({
+      path: sourcePath,
+      content,
+    });
+  }
```

Comment:
I guess this is passing for all cases now, but shouldn't this be actually getting the sources from the filesystem instead of assuming the paths in the metadata file?

Ie. here you are looking at the file paths in metadata.sources field. This assumes the paths in the metadata structure is the same as in the tests. But this doesn't has to be the case since we actually check the files exist with their hashes and not their paths. So the "correct" way would be to recursively traverse the "sources" folder and add every source file to the sources object.

Not a big issue but to me this feels to be the right way.

Reply:
I see your point. It was just to keep the previous sources structure and I thought in this way it would work. What do you think it's best to do here? Maybe we'll do a final iteration on this tests util details later?